src/helpers/json.ts

```diff
@@ -13,22 +15,26 @@ export function replacer(_: string, value: any) {
 
 export function reviver(_: string, value: any) {
   if (typeof value === "object" && value !== null) {
-    if (value.dataType === Map.name) {
+    if (value.dataType === mapDataType) {
       return new Map(value.value);
     } else if (
       value instanceof Object &&
       value.type === "Buffer" &&
       value.data
     ) {
       return Buffer.from(value.data);
-    } else if (value.dataType === AnonymousIdentity.name) {
-      return new AnonymousIdentity();
-    } else if (value.dataType === WebAuthnIdentity.name) {
-      return new WebAuthnIdentity(
-        value.cosePublicKey,
-        base64ToArrayBuffer(value.rawId)
-      );
+    } else {
+      switch (value.dataType) {
+        case AnonymousIdentity.dataType:
+          return new AnonymousIdentity()
```

Comment:
I'd like to make this backwards-compatible while allowing us to not be totally beholden to the previous mangling in local storages. Maybe something like:

```ts
} else if (anonymousTypes.includes(value.dataType)) {
  return new AnonymousIdentity()
}  else if (webauthnTypes.includes(value.dataType)) {
  return new WebAuthnIdentity(...)
}
```
Where `anonymousTypes` is something like `["r", "AnonymousIdentity", ...]`.